world/petitions/models.py

```diff
@@ -10,6 +10,93 @@
 
 from server.utils.exceptions import PayError
 from .exceptions import PetitionError
+from web.character.models import (Clue, SearchTag)
+from world.dominion.models import PlayerOrNpc
+from world.dominion.general_dominion_commands import CmdPatronage
+
+
+class Interest(SharedMemoryModel):
+    name = models.CharField(max_length=30)
+    wantedads = models.ManyToManyField("WantedAd", related_name="wanted_ads")
+
+
+class WantedAd(SharedMemoryModel):
+    """This is an advertisement for wanting an RP partner"""
+    MARRIAGE = 0
+    SPONSOR = 1
+    PROTEGE = 2
+    PLOT = 3
+    TIME = 4
+    MATCH_TYPES = ((MARRIAGE, "Marriage"), (SPONSOR, "Sponsor"),
+                   (PROTEGE, "Protege"), (PLOT, "Plot"), (TIME, "Time"))
+    playtime_start = models.PositiveIntegerField(default=0)
+    playtime_end = models.PositiveIntegerField(default=0)
+    prefer_weekend = models.BooleanField(default=False)
+    interests = models.ManyToManyField(Interest, blank=True)
+    clues = models.ManyToManyField(Clue, blank=True)
+    searchtags = models.ManyToManyField(SearchTag, blank=True)
+    blurb = models.TextField()
+    marriage = models.BooleanField(default=False)
+    sponsor = models.BooleanField(default=False)
+    protege = models.BooleanField(default=False)
+    plot = models.BooleanField(default=False)
+    active = models.BooleanField(default=False)
+    owner = models.OneToOneField(
+        PlayerOrNpc,
+        related_name="wanted_ad",
+        on_delete=models.CASCADE)
```

Comment:
This looks good. Intuitively I would have thought of it as a many-to-one with the booleans being a choices field, but this seems sensible

Reply:
I thought about using choices but when you're dealing with non-exclusive options like this I think a bunch of booleans are the easiest way to handle them.